lib/github_authentication/git_credential_helper.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module GithubAuthentication
+  class GitCredentialHelper
+    def initialize(pem:, installation_id:, app_id:, storage: nil)
+      @pem = pem
+      @installation_id = installation_id
+      @app_id = app_id
+      @storage = storage
+    end
+
+    def handle_get
+      description = parse_stdin
+
+      unless description['protocol'] == 'https' && description['host'] == 'github.com'
+        warn("Unsupported description: #{description}")
+        return 2
+      end
+
+      puts("password=#{provider.token(seconds_ttl: min_cache_ttl)}")
+      puts('username=api')
+
+      0
+    end
+
+    private
+
+    def min_cache_ttl
+      # Tokens are valid for 60 minutes, allow a 10 minute buffer at the start of jobs
```

Comment:
[nit] It's not always job, is it?

```suggestion
      # Tokens are valid for 60 minutes, allow a 10 minute buffer
```